### ISSUE TITLE
[Quarkus 2.5.1.Final] Camel exception when using jsonpath after unmarshalling in quarkus native

### ISSUE THREAD

**Issue by juangon** [apache/camel-quarkus#3369]
When using jsonpath inside a Camel RouteBuilder on Quarkus NATIVE mode (it works fine in JVM mode),  it throws this exception:

```
Caused by: org.apache.camel.CamelExchangeException: Cannot read message body as supported JSON value. Exchange[14440DAD29538A6-000000000000008D]
	at org.apache.camel.jsonpath.JsonPathEngine.doRead(JsonPathEngine.java:208)
	at org.apache.camel.jsonpath.JsonPathEngine.read(JsonPathEngine.java:105)
	at org.apache.camel.jsonpath.JsonPathExpression.evaluateJsonPath(JsonPathExpression.java:190)
	at org.apache.camel.jsonpath.JsonPathExpression.evaluate(JsonPathExpression.java:144)
```

It looks like it actually (silently) fails in the previous step when unmarshalling. Code:

```java
@ApplicationScoped
 public class CamelRouteBuilder extends RouteBuilder {
.
.
    @Override
    public void configure() throws Exception {
.
.
    from("direct:writeToKafka")
            .routeId("writeToKafka")
            .setBody(exchangeProperty("tickets"))
            //.unmarshal().json(JsonLibrary.Jackson)
            .to("dataformat:json-jackson:unmarshal?unmarshalType=com.example.TicketResponse")
            .split().jsonpath("$.result[*]")
```

camel quarkus dependencies:

```xml
<!--CAMEL QUARKUS-->
    <dependency>
      <groupId>org.apache.camel.quarkus</groupId>
      <artifactId>camel-quarkus-timer</artifactId>
    </dependency>
    <dependency>
      <groupId>org.apache.camel.quarkus</groupId>
      <artifactId>camel-quarkus-log</artifactId>
    </dependency>
    <dependency>
      <groupId>org.apache.camel.quarkus</groupId>
      <artifactId>camel-quarkus-microprofile-health</artifactId>
    </dependency>
    <dependency>
      <groupId>org.apache.camel.quarkus</groupId>
      <artifactId>camel-quarkus-microprofile-metrics</artifactId>
    </dependency>
    <dependency>
      <groupId>org.apache.camel.quarkus</groupId>
      <artifactId>camel-quarkus-http</artifactId>
    </dependency>
    
    <dependency>
      <groupId>org.apache.camel.quarkus</groupId>
      <artifactId>camel-quarkus-kafka</artifactId>
    </dependency>    

    <dependency>
      <groupId>org.apache.camel.quarkus</groupId>
      <artifactId>camel-quarkus-jackson</artifactId>
    </dependency>    

     <dependency>
      <groupId>org.apache.camel.quarkus</groupId>
     <artifactId>camel-quarkus-dataformat</artifactId>
    </dependency>  

    <dependency>
      <groupId>org.apache.camel.quarkus</groupId>
      <artifactId>camel-quarkus-direct</artifactId>
    </dependency>  

    <dependency>
      <groupId>org.apache.camel.quarkus</groupId>
      <artifactId>camel-quarkus-jsonpath</artifactId>
    </dependency>  

```

I already registered the POJO for reflection:

```java
@RegisterForReflection
public class TicketResponse implements Serializable {
  
   @JsonProperty(value = "result")
    private List<Ticket> result;

    public List<Ticket> getResult() {
        return result;
    }

    public void setResult(List<Ticket> result) {
        this.result = result;
    }    

}
```
```java
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(value = Include.NON_NULL)
@RegisterForReflection
public class Ticket implements Serializable {
.

.
    @JsonProperty("opened_at")
    @JsonDeserialize(using = DateDeserializer.class)
    private LocalDateTime createdAt;

}
```

```java

@RegisterForReflection
public class DateDeserializer extends JsonDeserializer<LocalDateTime> {
```

Using Quarkus 2.5.1.Final.

Thanks very much in advance!

**Comment by zhfeng** [comment]
@juangon are you trying with the lastest codebase ? and I think it has been upgraded to Quarkus 2.6.0.CR1. And if you have a simple reproducer to show this issue in native mode, I'd like to take a look.

**Comment by juangon** [comment]
Thanks for answering @zhfeng . I tried with latest 2.6.0.CR1 and still same problem. I am very busy nowadays, hopefully I can build a reproducer

**Comment by zbendhiba** [comment]
@juangon Can you gives up a heads-up. Is this still failing. If yes, could you please give a reproducer.
Thanks in advance

**Comment by juangon** [comment]
Sorry @zbendhiba   Going to close this issue as it seems it is fixed in 2.8.X.

Thanks very much

**Comment by liulitao-hyphen** [comment]
Big guy, have you found a solution?
